doc/sphinx/proof-engine/tactics.rst

```diff
@@ -3,35 +3,46 @@
-have to prove A and I have to prove B. This is backward reasoning from
-conclusion to premises. We say that the conclusion is the *goal* to
-prove and premises are the *subgoals*. The tactics implement *backward
-reasoning*. When applied to a goal, a tactic replaces this goal with
-the subgoals it generates. We say that a tactic reduces a goal to its
-subgoal(s).
-
-Each (sub)goal is denoted with a number. The current goal is numbered
-1. By default, a tactic is applied to the current goal, but one can
-address a particular goal in the list by writing n:tactic which means
-“apply tactic tactic to goal number n”. We can show the list of
-subgoals by typing Show (see Section :ref:`requestinginformation`).
-
-Since not every rule applies to a given statement, not every tactic can
-be used to reduce a given goal. In other words, before applying a tactic
-to a given goal, the system checks that some *preconditions* are
-satisfied. If it is not the case, the tactic raises an error message.
-
-Tactics are built from atomic tactics and tactic expressions (which
-extends the folklore notion of tactical) to combine those atomic
-tactics. This chapter is devoted to atomic tactics. The tactic
-language will be described in Chapter :ref:`ltac`.
+Tactics specify how to transform the :term:`proof state` of an
+incomplete proof to eventually generate a complete proof.
+
+Proofs can be developed in two basic ways: In :gdef:`forward reasoning`,
+the proof begins by proving simple statements that are then combined to prove the
+theorem statement as the last step of the proof. With forward reasoning,
+for example,
+the proof of `A /\\ B` would begin with proofs of `A` and `B`, which are
+then used to prove `A /\\ B`.  Forward reasoning is probably the most common
+approach in human-generated proofs.
+
+In :gdef:`backward reasoning`, the proof begins with the theorem statement
+as the goal, which is then gradually transformed until every subgoal generated
+along the way has been proven.  In this case, the proof of `A /\\ B` begins
+with that formula as the goal.  This can be transformed into two subgoals,
+`A` and `B`, followed by the proofs of `A` and `B`.  Coq and its tactics
+use backward reasoning.
+
+Tactics may fully prove a goal, in which case the goal is removed
+from the proof state.
+More commonly, tactics replace a goal with one or more subgoals.
+(We say that a tactic reduces a goal to its subgoals.)
+
+Most tactics require specific elements or preconditions to reduce a goal;
+they display error messages if they can't be applied to the goal.
+A few tactics, such as :tacn:`intros`, don't fail even if the proof state
+is unchanged.
+
+Goals are identified by number. The current goal is number
+1. Tactics are applied to the current goal by default.  They can
+be applied to another goal or to multiple goals with a
+:ref:`goal selector <goal-selectors>` such as :n:`2 : @tactic`.
+Goals can also be *shelved* (and later *unshelved*), so they are no
```

Comment:
We should probably define these terms (but probably later).

Reply:
Now works for me.  It's only a sentence or two.

I thought what I had was more or less a definition, though maybe could be a little more explicit.  What would you add?  I think the only goal category I don't mention here is "abandoned", which seems not important enough to mention in introductory material.  Indeed, it's not _really_ necessary to mention shelved goals here, either.

Reply:
Indeed, I agree that it's not really useful to mention shelved goals here. Shelved goals and given-up goals should be documented in https://coq.gitlab.io/-/coq/-/jobs/932404478/artifacts/_install_ci/share/doc/coq/sphinx/html/proofs/writing-proofs/proof-mode.html#postponing-the-proof-of-some-goals